File: backend/src/main/java/hu/bsstudio/raktr/dto/RentItemWithRentData.java
package hu.bsstudio.raktr.dto;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import hu.bsstudio.raktr.model.Rent;
import hu.bsstudio.raktr.model.RentItem;
import lombok.AllArgsConstructor;

@JsonSerialize
@JsonDeserialize
@AllArgsConstructor
public final class RentItemWithRentData {
    public RentItem rentItem;

    public Rent rent;
}

File: frontend-old/src/app/_model/RentItemWithRentData.ts
import {RentItem} from './RentItem';
import {Rent} from './Rent';

export class RentItemWithRentData {
    rentItem: RentItem;
    rent: Rent;

    static fromJson(rentItemWithRentData: RentItemWithRentData): RentItemWithRentData {
        return new RentItemWithRentData(
            RentItem.fromJson(rentItemWithRentData.rentItem),
            Rent.fromJson(rentItemWithRentData.rent)
        );
    }

    constructor(rentItem: RentItem, rent: Rent) {
        this.rentItem = rentItem;
        this.rent = rent;
    }
}
